### PR TITLE
Ft docs subdomain

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- Adds subdomain mappings for the docs
- Adds applying of nginx ingresses to GitHub actions

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
    - [AN-62](https://airqoteam.atlassian.net/browse/AN-62?atlOrigin=eyJpIjoiMTdmMzdlOTIwNmI3NGM0NWEzM2MyN2JlNWM2MWFkY2YiLCJwIjoiaiJ9)
    - [AN-107](https://airqoteam.atlassian.net/browse/AN-107?atlOrigin=eyJpIjoiZDFkZWU1ZWQyNTMyNGIyMWFiMTZiZjhjMjM1MjNjMTEiLCJwIjoiaiJ9)

**_HOW DO I TEST OUT THIS PR?_**
- Study the raw code.
- Checkout the docs endpoint below.

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- [Docs](http://34.77.118.157:31104/#/)

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- n/a

**_ARE THERE ANY RELATED PRs?_**
    - [This frontend PR](https://github.com/airqo-platform/AirQo-frontend/pull/807)

### CODE REVIEW

File: k8s/nginx-ingress/prod-platform-ingress-resource.yaml [hunk 1]
```diff
@@ -111,6 +111,16 @@ spec:
                 name: airqo-calibrate-app-svc
                 port: 
                   number: 80
+    - host: docs.airqo.net
```
Comment (123MwanjeMike) [inline k8s/nginx-ingress/prod-platform-ingress-resource.yaml:114]: [docs custom subdomain](https://github.com/airqo-platform/AirQo-api/pull/1073/commits/81c31513896e39a498bb2bafe08cdc3541ea67d3)

File: k8s/nginx-ingress/stage-platform-ingress-resource.yaml [hunk 2]
```diff
@@ -31,6 +31,16 @@ spec:
                 name: airqo-stage-calibrate-app-svc
                 port: 
                   number: 80
+    - host: staging-docs.airqo.net
```
Comment (123MwanjeMike) [inline k8s/nginx-ingress/stage-platform-ingress-resource.yaml:34]: [docs staging custom subdomain](https://github.com/airqo-platform/AirQo-api/pull/1073/commits/81c31513896e39a498bb2bafe08cdc3541ea67d3)

File: .github/workflows/deploy-apis-to-staging.yml [hunk 3]
```diff
@@ -1566,4 +1573,23 @@ jobs:
...
+        uses: actions/checkout@v2
+
+      - name: Login to K8S
+        uses: azure/k8s-set-context@v1
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.K8S_CONFIG_STAGE }}
+
+      - name: Apply ingress changes
+        run: kubectl apply -f k8s/nginx-ingress/stage-platform-ingress-resource.yaml
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-staging.yml:1595]: [add apply staging nginx ingress to workflow](https://github.com/airqo-platform/AirQo-api/pull/1073/commits/5b460a36a5d7e44425a2c79840008e6294fbcf3c)

File: .github/workflows/deploy-apis-to-production.yml [hunk 4]
```diff
@@ -1504,4 +1511,25 @@ jobs:
...
+
+      - name: Login to K8S
+        uses: azure/k8s-set-context@v1
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.K8S_CONFIG_PROD }}
+
+      - name: Apply ingress changes
+        run: |
+          kubectl apply -f k8s/nginx-ingress/prod-api-ingress-resource.yaml
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1534]: [add apply production nginx ingress to workflow](https://github.com/airqo-platform/AirQo-api/pull/1073/commits/49adbb6223196a994b6acaa359aef2e411d4de00)

File: k8s/nginx-ingress/prod-platform-ingress-resource.yaml (hunk 1 above)
Comment (NoahNsimbe) [reply k8s/nginx-ingress/prod-platform-ingress-resource.yaml:114]: hi @123MwanjeMike. So we are hosting all the AirQo documentation (Engagements, partnerships, resources) on GitHub or just the AirQo API docs.

Comment (123MwanjeMike) [reply k8s/nginx-ingress/prod-platform-ingress-resource.yaml:114]: Thanks @NoahNsimbe 
We are going to use it to host all the public information that includes the API and resources.
I am not sure if we shall also add the engagements and Partnerships, but @Baalmart can confirm.

Comment (123MwanjeMike) [reply k8s/nginx-ingress/prod-platform-ingress-resource.yaml:114]: The Internal docs will remain in Postman as per now.

Comment (Baalmart) [reply k8s/nginx-ingress/prod-platform-ingress-resource.yaml:114]: engagements and partnerships can also reside here if we wish to. Just an effort to add version control to all our content. Thanks @123MwanjeMike for taking the initiative and acting on it! Amazing stuff! :)

**Review by Baalmart — COMMENTED** [review]
All LGTMA right now.
Hi @123MwanjeMike , pleas share the links to:

- staging
- production.

The ones you shared in the PR just lead to the PR. :)

**Comment by 123MwanjeMike** [comment]
> All LGTMA right now. Hi @123MwanjeMike , pleas share the links to:
> 
>     * staging
> 
>     * production.
> 
> 
> The ones you shared in the PR just lead to the PR. :)

Hi @Baalmart 
The changes haven't been added to production.
However, staging, for [this link](http://34.77.118.157:31104/#/) for a sample.

File: k8s/nginx-ingress/prod-platform-ingress-resource.yaml (hunk 1 above)
Comment (123MwanjeMike) [reply k8s/nginx-ingress/prod-platform-ingress-resource.yaml:114]: Great, thanks @Baalmart

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Pull request merged**